lib/commands/touch.js

```diff
@@ -1,9 +1,51 @@
 const commands = {};
-
 //This is needed to make clicks on -image elements work properly
 commands.performActions = async function (actions) {
-  return await this.winAppDriver.sendCommand('/actions', 'POST', {actions});
+  switch (actions[0].id) {
+    case 'default keyboard':
+      this.log.info('Mapping keyboard actions');
+      return await performKeyboardCommands(this.winAppDriver, actions[0].actions);
+    case 'default mouse':
+      this.log.info('Mapping mouse actions');
+      return await performMouseCommands(this.winAppDriver, actions[0].actions, this.log);
+    default:
+      this.log.info('Going default actions route');
+      return await this.winAppDriver.sendCommand('/actions', 'POST', {actions});
+  };
 };
 
+async function performKeyboardCommands(winAppDriver, actions) {
+  const keysFormatted = formatKeyboardInput(actions);
+  return await winAppDriver.sendCommand('/keys', 'POST', {'value': keysFormatted });
+}
+
+async function performMouseCommands(winAppDriver, actions, log) {
+  let result;
+  for (let action in actions) {
+    switch (actions[action].type) {
+      case 'pointerMove':
+        result = await winAppDriver.sendCommand('/moveto', 'POST', {'element': actions[action].origin.ELEMENT });
+        break;
+      case 'pointerDown':
+        result = await winAppDriver.sendCommand('/buttondown', 'POST', {});
+        break;
+      case 'pointerUp':
+        result = await winAppDriver.sendCommand('/buttonup', 'POST', {});
+        break;
+      default:
+        log.info('No mouse action mapped for type: ' + JSON.stringify(actions[action].type) + ' Moving to next action');
+    }
+  };
+  return result;
+}
+
+function formatKeyboardInput(actions) {
+  const keys = new Set();
```

Comment:
Set is unordered. 
Also, I am not sure this formatter would work as expected. Key blocks in w3c actions may include delays or depress /release special keys while other are being pressed. The current logic does ignore such scenarios

Reply:
The way the /keys endpoint from winAppDriver works is that it presses all the given keys simultaneously and then releases them. 
the sendKeys method under Actions creates 2 actions for each key (1 press and 1 release). We don't need to specify to the /keys endpoint what we want it to to with the keys, we just need to pass keys to it.
I tested multiple scenarios like combinations of keys (SHIFT + "a") etc and it works like using the Keyboard interface directly.

Reply:
Keys might work differently in w3c, that's true..but currently I would say that we need sendKeys to work like it did before and this should be achieved. The /keys endpoint from WinAppDriver doesn't have support for anything extra

Reply:
Gave up on using set due to unordered issue, just iterating over values now

Reply:
you may use lodash's https://lodash.com/docs/2.4.2#uniq method, which preserves values order. Also, we probably should log somewhere the parsed sequence to make it clear it is going to be simplified. Actions allow to create much more complicated key sequences and here we are deliberately simplifying these

Reply:
It won't do what's expected. Set was a wrong pick from the start.
When I perform sendKeys for a char thrugh Actions interface, that is converted by default to 2 actions (key press and key release for the same key). I used a set to avoid these duplicated because the windAppDriver endpoint needs only the key values and it does the press/release behind it.
But using set would also remove duplicated keys, so sendKeys("aa") would have just types "a".
This new approach has this fixed